dds/DCPS/XTypes/MemberDescriptor.cpp

```diff
@@ -8,52 +8,241 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+{
+  name_ = val;
+}
+
+const char* MemberDescriptorImpl::name() const
+{
+  return name_.c_str();
+}
+
+void MemberDescriptorImpl::id(const DDS::MemberId id)
+{
+  id_ = id;
+}
+
+DDS::MemberId MemberDescriptorImpl::id() const
+{
+  return id_;
+}
+
+void MemberDescriptorImpl::type(DDS::DynamicType_ptr type)
+{
+  type_ = type;
+}
+
+DDS::DynamicType_ptr MemberDescriptorImpl::type() const
+{
+  return type_;
+}
+
+void MemberDescriptorImpl::default_value(char* val)
+{
+  default_value_ = val;
+}
+
+void MemberDescriptorImpl::default_value(const char* val)
+{
+  default_value_ = val;
+}
+
+void MemberDescriptorImpl::default_value(const ::CORBA::String_var &val)
+{
+    default_value_ = val;
+}
+
+const char* MemberDescriptorImpl::default_value() const
+{
+  return default_value_.c_str();
+}
+
+void MemberDescriptorImpl::index(const ::CORBA::ULong index)
+{
+  index_ = index;
+}
+
+::CORBA::ULong MemberDescriptorImpl::index() const
+{
+  return index_;
+}
+
+void MemberDescriptorImpl::label(const DDS::UnionCaseLabelSeq& label)
+{
+  label_ = label;
+}
+
+const DDS::UnionCaseLabelSeq& MemberDescriptorImpl::label() const
+{
+  return label_;
+}
+
+DDS::UnionCaseLabelSeq& MemberDescriptorImpl::label()
+{
+  return label_;
+}
+
+void MemberDescriptorImpl::try_construct_kind(DDS::TryConstructKind kind)
+{
+  try_construct_kind_ = kind;
+}
+
+DDS::TryConstructKind MemberDescriptorImpl::try_construct_kind() const
+{
+  return try_construct_kind_;
+}
+
+void MemberDescriptorImpl::is_key(const ::CORBA::Boolean is_key)
+{
+  is_key_ = is_key;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_key() const
+{
+  return is_key_;
+}
+
+void MemberDescriptorImpl::is_optional(const ::CORBA::Boolean is_optional)
+{
+  is_optional_ = is_optional;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_optional() const
+{
+  return is_optional_;
+}
+
+void MemberDescriptorImpl::is_must_understand(const ::CORBA::Boolean is_must_understand)
+{
+  is_must_understand_ = is_must_understand;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_must_understand() const
+{
+  return is_must_understand_;
+}
+
+void MemberDescriptorImpl::is_shared(const ::CORBA::Boolean is_shared)
+{
+  is_shared_ = is_shared;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_shared() const
+{
+  return is_shared_;
+}
+
+void MemberDescriptorImpl::is_default_label(const ::CORBA::Boolean is_default_label)
+{
+  is_default_label_ = is_default_label;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_default_label() const
+{
+  return is_default_label_;
+}
+
+DDS::ReturnCode_t MemberDescriptorImpl::copy_from(DDS::MemberDescriptor*)
+{
+  // FUTURE
+  return DDS::RETCODE_UNSUPPORTED;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::is_consistent()
+{
+  // FUTURE
+  ACE_ERROR((LM_ERROR, "(%P|%t) ERRROR: MemberDescriptorImpl::is_consistent is not implemented\n"));
+  return false;
+}
+
+::CORBA::Boolean MemberDescriptorImpl::_tao_marshal__DDS_MemberDescriptor(TAO_OutputCDR &, TAO_ChunkInfo &) const
```

Comment:
Since these should never be marshalled, I don't think we need to do anything.  Maybe print an error.